wallet-api/tutorial/Tutorial.md

```diff
@@ -213,7 +220,7 @@ In the `Collect` case, the current slot must be between `deadline` and `collecti
 
 **Note (Operators in On-Chain Code)** We can use the operators `>`, `<`, `>=`, `<=` and `==` to compare `Int` values in PLC without having to define them in the script itself, as we did with `&&`. The compiler plugin that translates Haskell Core to Plutus Core knows about those operators because `Int` is a primitive type in Plutus Core and operations on it are built in. `Bool` on the other hand is treated like any other user-defined data type, and all functions that operate on it must be defined locally. More details can be found in the [PlutusTx tutorial](../../plutus-tx/tutorial/Tutorial.md).
 
-Finally, we can return the unit value `()` if `isValid` is true, or fail with an error otherwise.
+Finally, we can return the unit value `()` if `isValid` is true, or fail with an error otherwise. We can produce an error using the `P.error` function from the Prelude, but the signature of `P.error` is `forall a. () -> a` and therefore we alway have to apply it to a unit value. `P.error` is different from Haskell's `undefined` (of type `forall a. a`) because of differences in the type systems of the two languages. A more in-depth explanation can be found [here](../../plutus-tx-plugin/src/Language/PlutusTx/Compiler/ValueRestriction.hs). 
```

Comment:
Not sure about linking people to source files - I do not promise that that is a stable source of documentation!

Reply:
Ok, I removed the link.